rotors_gazebo_plugins/src/gazebo_odometry_plugin.cpp

```diff
@@ -112,6 +112,64 @@ void GazeboOdometryPlugin::Load(physics::ModelPtr _model, sdf::ElementPtr _sdf)
   if (parent_link_ == NULL && parent_frame_id_ != kDefaultParentFrameId) {
     gzthrow("[gazebo_odometry_plugin] Couldn't find specified parent link \"" << parent_frame_id_ << "\".");
   }
+
+#if SDF_MAJOR_VERSION >= 3
```

Comment:
To keep maintainability I would prefer if we add the `.X()` accessors to the `sdf` vector versions.
In the header, something like this (untested) should do:

```C++
#define Vector3 Vector3Sdf
// Include for Vector3
#undef Vector3

class Vector3 : Vector3Sdf {
 public:
  void X() {return this->x;};
};
```

Not sure if this really works, but in my opinion this would be a lot nicer, as we don't have code duplication.